### DISCUSSION TITLE
Migrating from plug-and-play to custom auth

### DISCUSSION THREAD

**Discussion by Kahtaf** [Web3Auth/Web3Auth#421] [Help]
Is there a migration path from [plug-and-play](https://web3auth.io/docs/sdk/web/web3auth/) to [custom auth](https://web3auth.io/docs/sdk/web/core/custom-authentication)? Will users who have signed in with plug-and-play (Google + openlogin) previously, be able to sign in with custom auth (our own Google client ID and verifier), and use the same torus wallet as before?

**Comment by yashovardhan** [comment]
Hey Kahtaf,

Thanks for your question.The concept will remain the same for both Plug and Play UI (`@web3auth/web3auth`) and Play and Play Core(`@web3auth/core`) SDKs. This is because Plug and Play UI is just a sub class of Plug and Play Core. Hence if you just directly want to migrate you can do so by changing the following functions:
- Instantiation - change the extended `Web3Auth` class to `Web3AuthCore`
- Initialisation - change the `initModal` function to `init` function
- Logging in - change the `connect` function to `connectTo` function

Rest depending upon the arguments, you can have a look at our SDK References and figure out a setting of your choice.


Further, coming to your question about Custom Authentication and how will it affect the keys. A unique key is generated for each combination of a `verifier` and `clientId`. If you're not using custom authentication, then your keys will be the same in both SDKs. However, if you have a new verifier setup for enabling custom authentication, it will change the users' keys. There no direct way of shifting your users directly to a new custom authentication setup. They'll have a new account generated from Web3Auth. However there might be some hacks around this to import users' existing keys. 

Also one thing to note, they won't have a torus wallet account. Web3Auth account is scoped to your dapp. Torus Wallet accounts are scoped to Torus Wallet. Think about them as two different dapps implementing Web3Auth with different client IDs. In terms of interoperability, the flow will be similar with different keys as mentioned before.
